specification/search/data-plane/Microsoft.Azure.Search.Service/preview/2017-11-11/examples/SearchServiceCreateDataSource.json

```diff
@@ -0,0 +1,52 @@
+{
+    "parameters": {
+        "searchServiceName": "myservice",
+        "api-version": "2017-11-11",
+        "dataSource": {
+            "name": "mydocdbdatasource",
+            "description": "My Cosmos DB data source.",
+            "type": "documentdb",
+            "credentials": {
+                "connectionString": "AccountEndpoint=https://myDocDbEndpoint.documents.azure.com;AccountKey=myDocDbAuthKey;Database=myDocDbDatabaseId"
+            },
+            "container": {
+                "name": "myDocDbCollectionId",
+                "query": "SELECT c.id, c.userId, tag, c._ts FROM c JOIN tag IN c.tags WHERE c._ts >= @HighWaterMark ORDER BY c._ts"
+            },
+            "dataChangeDetectionPolicy": {
+                "@odata.type": "#Microsoft.Azure.Search.HighWaterMarkChangeDetectionPolicy",
+                "highWaterMarkColumnName": "_ts"
+            },
+            "dataDeletionDetectionPolicy": {
+                "@odata.type": "#Microsoft.Azure.Search.SoftDeleteColumnDeletionDetectionPolicy",
+                "softDeleteColumnName": "isDeleted",
```

Comment:
These 2 properties are not defined in the type definition of "DataDeletionDetectionPolicy".

Reply:
This is a polymorphic type. Note the use of the `discriminator` property in `DataDeletionDetectionPolicy`. The value of `@odata.type` indicates that this object is of type `SoftDeleteColumnDeletionPolicy`. If you look at the definition of `SoftDeleteColumnDeletionPolicy` in the Swagger, you'll see the `softDeleteColumnName` and `softDeleteMarkerValue` properties are defined. Also note the use of `x-ms-discriminator-value` and `allOf` in `SoftDeleteColumnDeletionPolicy`.

Please ensure your validation tools understand polymorphism, as it is a crucial feature for modeling data plane APIs using AutoRest.